routing/missioncontrol_store.go

```diff
@@ -222,38 +238,103 @@ func deserializeResult(k, v []byte) (*paymentResult, error) {
 
 // AddResult adds a new result to the db.
 func (b *missionControlStore) AddResult(rp *paymentResult) error {
-	return kvdb.Update(b.db, func(tx kvdb.RwTx) error {
-		bucket := tx.ReadWriteBucket(resultsKey)
+	b.queueMx.Lock()
+	defer b.queueMx.Unlock()
+	b.queue.PushBack(rp)
 
-		// Prune oldest entries.
-		if b.maxRecords > 0 {
-			for b.numRecords >= b.maxRecords {
-				cursor := bucket.ReadWriteCursor()
-				cursor.First()
-				if err := cursor.Delete(); err != nil {
-					return err
+	return nil
+}
+
+// StopTicker stops the store goroutine.
+func (b *missionControlStore) StopTicker() {
+	close(b.done)
+	b.wg.Wait()
+}
+
+// RunTicker runs the store goroutine.
+func (b *missionControlStore) RunTicker() {
+	b.wg.Add(1)
+
+	go func() {
+		// TODO(bhandras): test multiple timeframes.
+		ticker := time.NewTicker(time.Second)
```

Comment:
Can we make this configurable to get rid of the TODO? At least up to `MissionControlConfig`, can always fully expose it as a command line/config file option later if needed.

Reply:
done